ludwig/typing.py

```diff
@@ -0,0 +1,28 @@
+"""Common typing for Ludwig dictionary parameters."""
```

Comment:
I'd like to imagine how this looks as a module in the future.  Once we define these types they will be imported throughout our code and user code as well.  Then, later on, we may replace these with dataclasses or something (which might be defined in their own files).

For that reason, I'd recommend to:
1.  call this module `types` instead of `typing`.      `from ludwig.types import ModelConfigDict`
AND
2. maybe import these into the `__init__.py`.       `from ludwig import ModelConfigDict`
Making these framework-level types like`torch.Tensor`

Reply:
I've gone with the first suggestion and will hold off on the second. There are other `types.py` files in the repo that could be worth considering more before the global unification.